### ISSUE TITLE
Bearer tokens should not be supported or be deprecated

### ISSUE THREAD

**Issue by Denisthemalice** [ietf-wg-gnap/gnap-core-protocol#289]
Bearer tokens are so easy to be forwarded by a client to another client that they should not be supported.

This does not mean that one client cannot forward sender-constrained tokens to another client, but this is less straightforward.

**Comment by jricher** [comment]
The attacks on bearer tokens don't involve a client knowingly forwarding the token, but rather another party such as the rs using or forwarding the token. A client can always simply forward the key material, or even make a call on behalf of someone else. Nothing can stop that. 

This was discussed extensively when the group was founded. There is strong desire to keep support for bearer tokens but GNAP has made them non-default and requires them to be flagged specially.

**Comment by aaronpk** [comment]
Agreed, a client can voluntarily share key material and tokens or even raw data, and there's no way to stop that. Like Justin said, the interesting things that can be prevented by avoiding bearer tokens are parties _other than the client_ (e.g. a Resource Server) passing on the token accidentally or maliciously. That said, there are still cases where this is not a concern, such as when all roles (client, RS, AS) are controlled by a single entity, in which case the simplicity of bearer tokens may be appealing.

**Comment by Denisthemalice** [comment]
@jricher wrote: 

> "The attacks on bearer tokens don't involve a client knowingly forwarding the token, but rather another party such as the rs using or forwarding the token."

This forwarding attack by a RS can easily be countered. It is sufficient to include into the access token either a true identifier or a concealed identifier of the RS, so that each RS can verify to which RS the access token was intended to. 

This has been explained on the mailing list about one year ago.

This is an additional argument why the structure of the access token should be defined in the core document.

@jricher also wrote: 

> A client can always simply forward the key material, or even make a call on behalf of someone else. **_Nothing can stop that_**.

Indeed, it is possible that a user forwards the key material to another, but the attack is less easy for the second user if the token is a sender-constrained token. However, the sentence "**_Nothing can stop that_**" is inappropriate. Collusion attacks between two users cannot be **_prevented_** but can be **_detected_** . If the RS is capable to perform that detection, then the attack can be defeated (and thus "stopped").

If a client simply makes a **_call on behalf of someone else_**, it will be get attributes and/or rights associated with that someone else.
It may impersonate that someone else but this does not mean that the attributes and rights from someone can be mixed with the attributes and rights from that someone else.

Details on how to defeat user collaborative attacks **_and their implications_** are discussed in issue #290.

**Comment by aaronpk** [comment]
@Denisthemalice please respond to me comment on #290 before continuing to repeat arguments in other issues

**Comment by Denisthemalice** [comment]
I have just replied to your comment on #290
